### PR TITLE
testing: Enable stack monitoring local testing

### PR DESCRIPTION
**Pull request by marclop**
## Motivation/summary

This PR adds a few things to make it easier to test things under the
Elastic Agent in a semi-automated manner, using `systemtest/cmd/runapm`
to create a new Elastic Agent policy with a locally built `apm-server`
binary.
Also adds the ability to enable stack monitoring locally via a new
`metricbeat` container that is only added when `docker-compose --profile
monitoring` is set, with two scripts to make this possible:

- `testing/docker/fleet-server/agent-monitoring.sh` is run inside the
container and modifies the agent configuration to enable monitoring and
publish the metrics in all interfaces.
- `testing/stack-monitoring.sh` can be run from your machine to start
the stack with Stack Monitoring enabled.

`TESTING.md` has been update with all the necessary instructions.

## Checklist

~- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)~
~- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)`~
- [x] Documentation has been updated

## How to test these changes

1. See the `TESTING.md` document.
2. Run `./testing/stack-monitoring.sh` and verify that Elasticsearch and APM Server show metrics in the Stack Monitoring UI.

## Related issues

None, but we want to improve our ability to test things locally.

### CODE REVIEW

File: TESTING.md [hunk 1]
```diff
@@ -35,3 +38,44 @@ $ git checkout main
...
+is working as expected.
+
+### Injecting an APM Server binary into Elastic Agent
+
+Since APM Server is now run by the Elastic Agent in managed mode, only testing the APM Server in Standalone mode will
+not completely test the supported and recommended APM Server setup. To reuse the `docker-compose.yml` components and
+ease testing, you can inject an `apm-server` binary in the Elastic Agent container so a locally built version can be
+tested while making changes to the APM Server or before a release is published.
+
+To do so, you can build the binary locally with and copy the apm-server binary to the folder that is bindmounted in the
```
Comment (stuartnelson3) [inline TESTING.md:62]: is this an extra "with"?

File: TESTING.md [hunk 2]
```diff
@@ -35,3 +38,44 @@ $ git checkout main
...
+```
+
+The workflow that needs to be followed is:
+
+1. Build the APM Server and copy it to the expected folder (As shown above).
+2. `docker-compose up -d`.
+3. Once everything is up and running, make sure the APM Integration is installed and assigned to the Fleet Policy.
+4. Restart the `fleet-server` container: `docker-compose restart fleet-server`.
+
+We need to restart the Elastic Agent container after the APM Server as extracted by the ELastic Agaent has been run
```
Comment (stuartnelson3) [inline TESTING.md:77]: `/s/ELastic/Elastic/`

File: docker-compose.yml [hunk 3]
```diff
@@ -98,3 +100,18 @@ services:
     volumes:
       - "./testing/docker/package-registry/config.yml:/package-registry/config.yml"
       - "./build/integrations/apm:/packages/local/apm"
+
+  metricbeat:
+    image: docker.elastic.co/beats/metricbeat:8.2.0-43df679f-SNAPSHOT
```
Comment (stuartnelson3) [inline docker-compose.yml:105]: will the version for this also be automatically updated when the other container versions in this file are updated?

**Review by stuartnelson3 — APPROVED** [review]
couple small typos and a question, looks good!

File: docker-compose.yml (hunk 3 above)
Comment (marclop) [reply docker-compose.yml:105]: I looked at the update script and I believe it will be as well, yes.

Comment (marclop) [reply docker-compose.yml]: just tested this locally and it seemed to update all the service images 🙂

**Review by axw — COMMENTED** [review]
Would it be possible and reasonable to use `systemtest/cmd/runapm`, instead of customising the fleet-server container's entrypoint?

**Review by simitt — APPROVED** [review]
Tested this quickly and everything worked as expected. Great improvement for enabling stack monitoring locally!

**Comment by marclop** [comment]
We can't merge this PR until the Jenkins workers docker-compose binaries have been updated.

File: .ci/scripts/linux-test.sh [hunk 4]
```diff
@@ -14,7 +14,8 @@ trap cleanup EXIT
 make update apm-server
 
 # Start docker-compose environment first, so it doesn't count towards the test timeout.
-docker-compose up -d
+tree ${PYTHON_ENV}
```
Comment (marclop) [inline .ci/scripts/linux-test.sh:17]:
```suggestion
```

File: .ci/scripts/linux-test.sh [hunk 5]
```diff
@@ -14,7 +14,8 @@ trap cleanup EXIT
 make update apm-server
 
 # Start docker-compose environment first, so it doesn't count towards the test timeout.
-docker-compose up -d
+ls -lRt ${PYTHON_ENV}
+${PYTHON_ENV}/linux/bin/docker-compose up -d
```
Comment (marclop) [inline .ci/scripts/linux-test.sh:18]:
```suggestion
${PYTHON_ENV}/build/ve/linux/bin/docker-compose up -d
```

File: .ci/scripts/linux-test.sh [hunk 6]
```diff
@@ -14,7 +14,7 @@ trap cleanup EXIT
 make update apm-server
 
 # Start docker-compose environment first, so it doesn't count towards the test timeout.
-docker-compose up -d
+${PYTHON_ENV}/build/ve/linux/bin/docker-compose up -d
```
Comment (marclop) [inline .ci/scripts/linux-test.sh:17]:
```suggestion
which docker-compose
${PYTHON_ENV}/build/ve/linux/bin/docker-compose up -d
export PATH=$PATH:${PYTHON_ENV}/build/ve/linux/bin
```

**Pull request merged**